folly/experimental/io/EpollBackend.cpp

```diff
@@ -94,7 +94,9 @@ void SignalRegistry::notify(int sig) {
   int fd = notifyFd_.load();
   if (fd >= 0) {
     uint8_t sigNum = static_cast<uint8_t>(sig);
-    ::write(fd, &sigNum, 1);
+    if (::write(fd, &sigNum, 1) != 1) {
+      throw std::runtime_error("Failed to write all the byes.");
+    }
```

Comment:
This looks like the type of `write` that is infallible, i.e., cannot fail (as long as the inputs are correct), and total, i.e. cannot be a partial write.

And if this `write` wre to fail, there is nothing that would know how to clean this up.

Failure would be an invariant violation. If we are looking for invariant violations, such as we might do in debug builds, we would use something in the family of `assert` (eg `DCHECK`, `PCHECK`, `FOLLY_SAFE_DCHECK`, etc).

Partial writes may occur if the backend has insufficient resources to accept the full write, and if the write is non-blocking or is interrupted by a signal while blocking. But this is not the sort of write where the backend could have insufficient resources.